add: МЭКи [IDB IGNORE]

NightDawnFox opened the pull request:
<!-- **ВАЖНО!!!** Если Ваш ПР содержит много .dmi файлов, которые могут создать излишнюю нагрузку на IconDiffBot2, тогда добавьте в название ПРа тэг [IDB IGNORE] -->
<!-- Например, "Добавил 1kk новых спрайтов [IDB IGNORE]" -->

<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- Список тегов для ПРа:
- add: Вы добавляете что-то новое.
- admin: Вы меняете что-то связанное с администрацией. (кнопки, управление, панели, щитспавн и т.д.)
- balance: Вы производите балансировку в игре.
- bugfix: Вы исправили некий баг.
- code_imp: Вы имплементируете новое для билда, не меняя при этом ничего в самой игре.
- config: Вы меняете конфиг или работу SQL.
- del: Вы удаляете что-то из игры.
- experiment: Ваш ПР сделан с целью какого-то эксперимента.
- map Вы меняете только карту.
- local Вы делаете добавляете новый текст на русском или переводите на русский.
- imageadd:  Вы добавили новый спрайт.
- imagedel:  Вы удалили старый спрайт.
- soundadd: Вы добавили новый звук.
- sounddel: Вы удалили старый звук.
- spellcheck: Вы исправили опечатку.
- tweak: Вы внесли незначительную правку.
- refactor: Вы полностью переписали старый код, улучшив его, НО не изменив функционал.
- server: Вы меняете что-то связанное с серверной частью или Github.
- wip: Ваш PR в драфте и планируется длительная разработка. -->

## Что этот ПР делает
- Добавляет новый предмет: генератор энергетического щита. Установка щита позволяет стрелять со стороны пользователя, при этом давая защиту от стрельбы в ответ.
- Для создания кларка теперь нужна обычная ева
- Небольшие рефакторы кода
- ну эээ еще пару звуков добавил разных и спрайтов.
<sub> ну и еще добавляет модсьюты </sub>

На текущем этапе это примерное восстановление ПРа рерика полуторалетней давности. Есть несколько очевидных проблем, которые будут решаться. До релиза еще примерно месяц минимум. Ну, по крайней мере оно компиллится, уже хорошо.
Небольшой список задач для выполнения:

- [x]  Ревизия перевода. Завершение перевода, итоговое решение по названию модсьютов и приведение перевода в единый стиль.

- [x] Ревизия кода. Текущий вариант кода - оффопаровский и скорее всего его лучше будет переписать под тг-шный, где это возможно.

- [x] Работа с тгуи. Текущий вариант, судя по всему, полностью не работает. Необходимо привести старую версию тгуи под текущий стандарт. Необходимо настроить отображение ру-неймсов в тгуи модсьютов.

- [x] Первичные тесты всех систем и модулей. Поиск очевидных и геймбрейкинг багов, и их исправление. Прогонка по затратам памяти (хотя ничего нагружать вроде не должно)

- [x] Внедрение модсьютов в игру и станцию. (на текущем этапе по всей станции находятся классические харды, это я пока-что не трогал. Апплинк тоже затронут не был, как и ОБР.)

- [x] Углубленное добавление модулей с ТГ, от которых отказалась оффопара. ( В особенности модули, завязанные на мульти-з)

- [ ] Добавление новых модсьютов исходя из нашей разницы в билдах (начиная с модсьюта контрактора и заканчивая тайпановским чудищем)

<!-- Опишите, что делает ваш пулл-реквест, укажите основные изменения. -->
<!-- В случае исправления бага, укажите ссылку на канал в #баг-репорты-v2 или issue в репозитории. В ином случае, опишите баг и ступени для его воспроизведения. -->

## Почему это хорошо для игры
Модсьюты > хардсьюты. Более продвинутая система замедления и модульности гораздо интереснее, чем текущие хардсьюты. Добавление новых уникальных особенностей модсьютам делается в сотни раз проще, чем для хардсьютов, где любая попытка что-то добавить превращается в говнокод.
<!-- Опишите, почему, по вашему, следует добавить эти изменения в игру. -->

## Демонстрация изменений
<img width="199" height="206" alt="image" src="https://github.com/user-attachments/assets/3af65232-d59a-4cdb-a204-39a1f489b57a" />
<img width="637" height="229" alt="image" src="https://github.com/user-attachments/assets/eea1991b-be30-45ab-a111-33f3e8e019fb" />
<img width="658" height="394" alt="image" src="https://github.com/user-attachments/assets/02160e05-f780-4cd4-86a3-eacdf6133909" />

<!-- Заполните, если Вы меняли спрайты, карту или ваши изменения требуют визуальной демонстрации изменений. -->

## Тестирование
ну оно запускается, уээээ
<!-- Как Вы тестировали свои изменения? Ведь тестировали? -->

PlayerUnknown14 commented:
МОДСЬЮТЫ НА ПАРАДИЗЕ ИС РИАЛ???

tsoivecen199015 commented:
> МОДСЬЮТЫ НА ПАРАДИЗЕ ИС РИАЛ???

Наёбка для уёбка

tsoivecen199015 commented:
А СТОП НЕ НАЁБКА

Anorak2024 commented:
<img width="506" height="154" alt="image" src="https://github.com/user-attachments/assets/8b5151bc-f1f1-4972-beb1-28fe9c0b1afe" />
Стоило начать что-то подозревать...
Специально номер красивый ждал?

NightDawnFox commented:
Номер кстати случайно совпал

L343-CJ commented:
Зюзя как всегда молодец :catsmile:

NightDawnFox committed:
спрайты

NightDawnFox committed:
Merge branch 'eds2' of https://github.com/NightDawnFox/Paradise into eds2

NightDawnFox committed:
фиксы унати

Ksaikok46 committed:
фикс

NightDawnFox committed:
Update mod_modules.dmi

Ksaikok46 committed:
nukes visible message on deploy/retract

Ksaikok46 committed:
правки перевода №2 + перевод + правки

NightDawnFox committed:
ломаем чексумы

Ksaikok46 committed:
правки перевода 3 + резня балунов

Ksaikok46 committed:
fix + core install balloon

NightDawnFox committed:
воксы - В С Ё

NightDawnFox committed:
спрайты - В С Ё

NightDawnFox committed:
ТЕПЕРЬ ТОЧНО ВСЁ

NightDawnFox committed:
фиксы

NightDawnFox committed:
небольшой багфикс

NightDawnFox committed:
мелкофиксы

NightDawnFox committed:
бля)

ROdenFL committed:
hat_stabilizer

ROdenFL committed:
Merge branch 'eds2' of https://github.com/NightDawnFox/Paradise into eds2

ROdenFL committed:
some fixes and deletions

NightDawnFox committed:
Проставлены req_slots на все модули

ROdenFL committed:
hydraulic + override

ROdenFL committed:
fixes

ROdenFL committed:
health

NightDawnFox committed:
аутфиты 1

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фиксобаги

NightDawnFox committed:
фиксы и кривой перевод

Ksaikok46 committed:
перевод мешков + фиксы перевода

ROdenFL committed:
final fixes

NightDawnFox committed:
Внедрение модсьютов в игру (1)

NightDawnFox committed:
до /obj/item/mod/module/pepper_shoulders

пиздец как же много я хуею

NightDawnFox committed:
уэээ

NightDawnFox committed:
фиксы

NightDawnFox committed:
вроде всё

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фикс1

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
ЕБАНЫЙ ДЭШ СУКА

NightDawnFox committed:
чистка дубликейт дефинитивс

ROdenFL committed:
prefinal fixes

ROdenFL committed:
linter

ROdenFL committed:
linter

ROdenFL committed:
linter

ROdenFL committed:
linter

ROdenFL committed:
linter

ROdenFL committed:
linter

ROdenFL committed:
linter

NightDawnFox committed:
Update _traits.dm

NightDawnFox committed:
linter

Ksaikok46 committed:
ещё перевод + правки

Добавил балуны, поправил кое-что тут и там

Ksaikok46 committed:
fix

PlayerUnknown14 committed:
Локализация modules_antag

PlayerUnknown14 committed:
Merge branch 'eds2' of https://github.com/nightdawnfox/paradise into pr/NightDawnFox/7500

PlayerUnknown14 committed:
Правки локализации модулей

NightDawnFox committed:
фиксы линтера

PlayerUnknown14 committed:
Правки локализации modules_engineering

NightDawnFox commented on code/modules/mod/modules/modules_engineering.dm:
```diff
@@ -369,11 +375,11 @@
 	switch(nozzle_mode)
 		if(EXTINGUISHER)
-			. += span_notice("Сейчас выбран режим огнетушителя.")
+			. += span_notice("Выбранный режим — <b>огнетушител</b>.")
```
ььь

PlayerUnknown14 replied:
ыыы

PlayerUnknown14 committed:
Правки локализации modules_general

PlayerUnknown14 committed:
Фикс

NightDawnFox committed:
карты

NightDawnFox committed:
Merge branch 'eds2' of https://github.com/NightDawnFox/Paradise into eds2

NightDawnFox committed:
бля про КМа забыл

NightDawnFox committed:
похуй пляшем

PlayerUnknown14 committed:
Merge branch 'master220' into eds2

PlayerUnknown14 committed:
Правки локализации modules_maint

PlayerUnknown14 committed:
Правки локализации modules_medical

PlayerUnknown14 committed:
Правки локализации modules_science

NightDawnFox committed:
фиксы за правками

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
тгуи

PlayerUnknown14 committed:
Правки за фиксами

PlayerUnknown14 committed:
Правки локализации modules_security

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
тгуи

PlayerUnknown14 committed:
Правки локализации modules_secrvice

PlayerUnknown14 committed:
Merge branch 'eds2' of https://github.com/nightdawnfox/paradise into pr/NightDawnFox/7500

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
по красоте

PlayerUnknown14 committed:
Правки локализации modules_supply

PlayerUnknown14 committed:
Правки локализации modules_visor

PlayerUnknown14 committed:
Локализация, табуляция, правки буковок.

PlayerUnknown14 committed:
Merge branch 'master220' into eds2

PlayerUnknown14 committed:
Правки гайда на добавление модсьютов

PlayerUnknown14 committed:
Правки локализации взаимодействия с МЭК (впереди только UI)

PlayerUnknown14 committed:
Юай-релейтед правки локализации (напиздюнькал, остались ещё темы модсьютов)

NightDawnFox committed:
фикс названий

NightDawnFox committed:
теперь точно фиксы

NightDawnFox committed:
забыл про вардена

PlayerUnknown14 committed:
Фикс

PlayerUnknown14 committed:
Фикс фикса

NightDawnFox committed:
Update modules_antag.dm

PlayerUnknown14 committed:
Правки mod_theme (почти доделал)

NightDawnFox committed:
фикс

NightDawnFox commented on code/modules/mod/mod_theme.dm:
```diff
@@ -488,20 +494,20 @@
+	extended_desc = "Разработка корпорации \"Нанотрейзен\", созданная на основе ранних индустриальных платформ \"Киберсан Индастриз\" с учётом \
+		обратной связи от полевых шахтёрских бригад. Модель \"Первопроходец\" оптимизирована для работы в астероидных шахтах и зонах с высоким уровнем \
+		термальных и взрывных рисков. Вместо традиционной брони использует аттракторы частиц пепла, формирующие временный абляционный слой, \
+		способный выдержать кратковременное экстремальное воздействие, но быстрой разрушающийся при физическом контакте. Ключевая особенность — \
```
быстрой - быстро

NightDawnFox commented on code/modules/mod/mod_theme.dm:
```diff
@@ -601,18 +607,18 @@
+		и транспортировки боеприпасов и тяжёлого снаряжения. Изначально произведённый для военной логистики, со временем костюм \
+		зарекомендовал себя эффективное и относительно доступное снаряжение для грузовых работ. \
+		Каркас выполнен из титанового сплава с распределённой гидравлической системой, многократно усиливающей грузоподъёмность оператора. \
+		Две дополнительные манипуляторные руки синхронизиуруются с нервной системой пользователя, \
```
синхронизиуруются

NightDawnFox commented on code/modules/mod/mod_theme.dm:
```diff
@@ -1817,7 +1844,7 @@
 	name = "засекреченной модели"
 	desc = "Высокотехнологичный, не существующий ни в одном документе бронированный костюм, созданный совместными усилиями \"Нанотрейзен\" и \"Стальной Гвардии\"."
-	extended_desc = "Громоздкий, словно танк, и легальный только в техническом смысле боевой костюм в зловещей черно-красной расцветке. \
+	extended_desc = "Громоздкий, словно танк, и легальный только в техническом смысле боевой костюм в зловещёй черно-красной расцветке. \
```
зловещёй???

PlayerUnknown14 committed:
Доправки mod_theme (ПЕРЕВОД — ВСЁ)

PlayerUnknown14 committed:
Merge branch 'eds2' of https://github.com/nightdawnfox/paradise into pr/NightDawnFox/7500

PlayerUnknown14 committed:
Merge branch 'master220' into eds2

PlayerUnknown14 committed:
Фиксы

PlayerUnknown14 committed:
Merge branch 'eds2' of https://github.com/nightdawnfox/paradise into pr/NightDawnFox/7500

NightDawnFox committed:
скин апплайер

NightDawnFox committed:
модкит (2)

NightDawnFox committed:
спрайты мода контрактника

NightDawnFox committed:
тема контрактора

PlayerUnknown14 committed:
Локализация темы МЭКа контрактника

NightDawnFox committed:
контрактор модуле

NightDawnFox committed:
Merge branch 'eds2' of https://github.com/NightDawnFox/Paradise into eds2

NightDawnFox committed:
модуль крюка (в щитспавн) и фиксы

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
странный спешл

NightDawnFox committed:
the man behind the man behind the man behind the man behind the m

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

PlayerUnknown14 committed:
Merge branch 'master220' into eds2

NightDawnFox committed:
улучшенный модуль

NightDawnFox committed:
Merge branch 'eds2' of https://github.com/NightDawnFox/Paradise into eds2

NightDawnFox committed:
переделка грапплинг хука

NightDawnFox committed:
два новых модуля улучшения

NightDawnFox committed:
Последние доделки

NightDawnFox committed:
балансная правка

NightDawnFox committed:
дубликате дефинионс

PlayerUnknown14 committed:
Правки локализации

PlayerUnknown14 committed:
И ещё микроправка

PlayerUnknown14 committed:
Merge branch 'master220' into eds2

PlayerUnknown14 committed:
Правки локализации

PlayerUnknown14 committed:
Правки локализации

PlayerUnknown14 committed:
Merge branch 'eds2' of https://github.com/nightdawnfox/paradise into pr/NightDawnFox/7500

PlayerUnknown14 committed:
Правки локализации

PlayerUnknown14 committed:
Замена макросов локализации на новые

PlayerUnknown14 committed:
Убрал лишний гендер

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update deployable.dm

SoulFN commented:
Тестмердж через час

PlayerUnknown14 committed:
Merge branch 'master220' into eds2

Antoonij commented on code/_onclick/click.dm:
```diff
@@ -265,6 +265,9 @@
 
 // See click_override.dm
 /mob/living/MiddleClickOn(atom/A)
+	. = SEND_SIGNAL(src, COMSIG_MOB_MIDDLECLICKON, A, src)
```
```suggestion
	. = SEND_SIGNAL(src, COMSIG_MOB_MIDDLECLICKON, A)
```
второй раз передавать себя не нужно. Тот, кому ты сигнал отправляешь - летит 1 аргументом в сигнальный прок

Antoonij commented on code/game/mecha/mech_fabricator.dm:
```diff
@@ -226,10 +228,15 @@
+			L.req_access = D.access_requirement
+			var/list/lockbox_access
+			for(var/access in L.req_access)
+				lockbox_access += "[get_access_desc(access)] "
```
```suggestion
			var/list/lockbox_access = list()
			for(var/access in L.req_access)
				lockbox_access += "[get_access_desc(access)] "
```
лист не инитнут, будет рантайм

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фикс 1

NightDawnFox committed:
фикс 2

NightDawnFox committed:
фикс 3

коммы

NightDawnFox committed:
фикс 4

NightDawnFox committed:
Last item in a long list should have a comma.

Last item in a long list should have a comma. Last item in a long list should have a comma. Last item in a long list should have a comma. Last item in a long list should have a comma. Last item in a long list should have a comma. Last item in a long list should have a comma. Last item in a long list should have a comma. Last item in a long list should have a comma. Last item in a long list should have a comma.

NightDawnFox committed:
дрифтинг контрактор апгрейд

NightDawnFox committed:
временные трудности

NightDawnFox committed:
фикс грее-драсков

NightDawnFox committed:
пара багфиксов

ИЛС меда снова отображается (как и все ИЛС)
модуль покачивания теперь работает
Теперь моды можно активировать и выключать лёжа. Отвертка теперь мгновенная.
Теперь дефибы отображаются нормально, и атом сей говорит не от людей, а от рукавиц (баг смешной, но всё равно)

NightDawnFox committed:
фиксобаги 2

Исправлены баги с модулем сферы - теперь его можно снять и емагнуть для стрельбы бомбами по станции.
Крюк больше не рикошетит

NightDawnFox committed:
Фиксобаги 3

модсьюты теперь активируются правильно у кукол с эквипом
Повышена комплексити у ригов ОБР и увеличен их лут
дефиб моду бригмеда
Фонари теперь тушатся абилками теней
клешня у мода карго теперь работает

NightDawnFox committed:
фиксобаги 4

модуль руды теперь работает
Кобура и кобура контрактника не ломаются при антидропе (наверное)
врины и манки больше не могут активировать моды. Блок еще можно носить

NightDawnFox committed:
но фан эллауд

NightDawnFox committed:
твики и багофиксы

1. Модуль хамелеона больше недоступен для нюки (упс)
2. Доступы у обшивок пишутся правильно
3. У МЭК класса страж теперь доступ оружейной
4. Теперь можно выдвигать и задвигать одиночные части МЭКа, когда он включен.
5. Теперь МЭКи защищают и хвост (от радиации к примеру)
6. МЭК класса "Путник" больше не скрывает маску и очки при выдвижении в выключенном виде
7. Модуль описания состояния в настройках пишет более правдивую информацию
8. Модуль пищеприемника теперь работает (Я НАДЕЮСЬ)
9. Все детали МЭКов переведены на русский язык (по принципу "Модуль чего-то там".
10. Перетасовка технологий - теперь все аномальные модули требуют 7 комбат, 7 инженерию, 7 бс, 7 плазму и 6 токсины. Понижение некоторых модулей в цене
11. Из щитспавна вытащен модуль замены органов - он позволяет хранить внутри костюма органы и агрессивно их менять с помощью стрельбы ими по куклам (НО ТОЛЬКО ПРИ ОПЕРАЦИИ).

NightDawnFox committed:
Большое обновление 1

1. Добавлен новый МЭК - МЭК пилота СБ. Быстро развертывается - броня чуть лучше, чем у картона, относительно быстрый
2. Всем МЭКам в режиме полного "экранирования" головы забрали возможность вытаскивать наушники
3. Обновление всех предсозданных МЭКов - теперь гораздо больше модулей заранее закреплено на панели для удобства пользователей.
4. В фабрикатор добавлены обшивки "нечастых" костюмов, спрятанные за технологиями. Все новые обшивки требуют технологии токсинов, а значит придется работать.
5. Магбутсы при включении теперь пишут о том, что он включены.

NightDawnFox committed:
фикс спрайтов

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
100 взрывозащиты у мода РД

ой

dageavtobusnick committed:
Merge branch 'master220' into eds2

dageavtobusnick committed:
compile fix

NightDawnFox committed:
юнит тест на модсьюты

NightDawnFox committed:
теперь точно включено

NightDawnFox committed:
фиксы

NightDawnFox committed:
Блядь

NightDawnFox committed:
пиздец

NightDawnFox committed:
иф фак

NightDawnFox committed:
Update mod_actions.dm

PlayerUnknown14 committed:
Правки локализации

PlayerUnknown14 committed:
Merge branch 'master220' into eds2

NightDawnFox committed:
фиксобаги

1. Убран шлем клоуна в хранилище его МЭКа (остатки)
2. Перенесены флаги одежды на уровень итемов (потому что у нас не только одежду можно надевать, но еще и вещи блядь)
3. Вырезан эффект ЭМИ у активного хамелеона
4. Исправлен баг с хамелеон-модулем
5. Добавлен оверлей света у рабочего фонарика
6. Вортекс дробовик больше не тратит энергию при не-стрельбе в пацифизме
7. Уборщики ОБР теперь носят МЭКи.

NightDawnFox committed:
фикс кинезиса

NightDawnFox committed:
Реворк аномалийных модулей

Теперь сила аномалийных ядер зависит от типа ядра внутри него.
Снижены технологий для аномалийных ядер

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
правки

NightDawnFox committed:
фиксы линтера

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фак ит шип ит

NightDawnFox committed:
рекомпиллить тгуи не бросим

NightDawnFox committed:
фикс контрактора

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фиксы совместимости

NightDawnFox committed:
вроде работает

NightDawnFox committed:
Update module.dm

NightDawnFox committed:
брбрбрбр

dageavtobusnick committed:
Merge branch 'master220' into eds2

dageavtobusnick commented on code/modules/wiremod/shell/module.dm:
```diff
@@ -251,7 +251,7 @@
 				modules_list += module.name
 		module_to_select.possible_options = modules_list
-		if (module_to_select.possible_options.len)
+		if(module_to_select.possible_options.len)
```
length(module_to_select.possible_options). Тут и на 280

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update paradise.dme

NightDawnFox committed:
наверное

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фикс неудачного решения конфликтов

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фиксы линтера

NightDawnFox committed:
увы!!

1. Добавлен модуль вспышек синдиката, установлен в синдириги
2. Экспериментальное удаление флага на сокрытие джампсьюта у всех модсьютов. Проверить, как будет выглядеть в игре.
3. Фикс возможности стрелять из вортекс шотгана

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
сорт ансортед

NightDawnFox committed:
новые модули

нерф энергощитов, модули генератора для мэков. и модуль миража в фабрикатор

NightDawnFox committed:
линтер

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Merge branch 'master220' into eds2

dageavtobusnick committed:
fix conflicts

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
фиксы

NightDawnFox committed:
фиксы крейтов

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
фикс

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update fast_load_multiz.dmm

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
conflict solve

dageavtobusnick committed:
Merge branch 'master220' into eds2

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

littleboobs commented:
/gemini review

littleboobs commented:
/gemini review

NightDawnFox commented:
Подсосал так подсосал, ничего не скажешь

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update gift.dm

littleboobs commented:
/gemini review

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
Выполнение приказов стюпид кланкера

ИИ насоветовал всего. Все что он насоветовал, я сделал.
Вроде даже работает

PlayerUnknown14 commented:
/gemini review

dageavtobusnick commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -90,12 +90,12 @@
 		internal = null
 		update_action_buttons_icon()
-
+	var/hide_hair_flags = HIDEHAIR|HIDEHEADHAIR|HIDEFACIALHAIR
```
Это лучше волбще дефайном

dageavtobusnick commented on code/modules/mod/modules/modules_engineering.dm:
```diff
@@ -458,11 +455,21 @@
+/obj/item/extinguisher/mini/mod/proc/regenerate_metal_charge()
 	metal_synthesis_charge++
 
+/obj/item/extinguisher/mini/mod/proc/move_nanofrost(obj/effect/nanofrost_container/A, atom/target, steps_left = 5)
```
Имя переменной

NightDawnFox committed:
правки от ИИ и второго ИИ

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
фиксы

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update fast_load_multiz.dmm

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update cyberiad.dmm

dageavtobusnick committed:
фикс проблемных хотспотов

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
zuza fix

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

NightDawnFox committed:
фикс магбутов

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7500

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick commented on code/__HELPERS/type2type.dm:
```diff
@@ -465,6 +465,11 @@ GLOBAL_LIST_INIT(modulo_angle_to_dir, list(NORTH,NORTHEAST,EAST,SOUTHEAST,SOUTH,
+/proc/text2bool(input)
+	if(!input)
+		return FALSE
+	return lowertext(input) in list("true", "1")
```
сделай однострочным макросом

dageavtobusnick commented on code/_onclick/hud/fullscreen.dm:
```diff
@@ -213,6 +213,62 @@
+	var/our_y = round(icon_y / world.icon_size)
+	given_turf = locate(owner.x + our_x - round(view_list[1] / 2), owner.y + our_y - round(view_list[2] / 2), owner.z)
+	given_x = round(icon_x - world.icon_size * our_x, 1)
+	given_y = round(icon_y - world.icon_size * our_y, 1)
```
У нас вроде как есть в билде дефайны для world.icon_size и "icon-x"

dageavtobusnick commented on code/controllers/subsystem/explosions.dm:
```diff
@@ -322,6 +322,9 @@ SUBSYSTEM_DEF(explosions)
 			helm_array.sense_explosion(x0, y0, z0, devastation_range, heavy_impact_range, light_impact_range, took, orig_dev_range, orig_heavy_range, orig_light_range)
+		if(istype(array, /obj/item/mod/module/reagent_scanner/advanced))
+			var/obj/item/mod/module/reagent_scanner/advanced/mod_array = array
+			mod_array.sense_explosion(x0,y0,z0,devastation_range,heavy_impact_range,light_impact_range,took,orig_dev_range,orig_heavy_range,orig_light_range)
```
```suggestion
			mod_array.sense_explosion(x0, y0, z0, devastation_range, heavy_impact_range, light_impact_range, took, orig_dev_range, orig_heavy_range, orig_light_range)
```

dageavtobusnick commented on code/datums/components/hat_stabilizer.dm:
```diff
@@ -0,0 +1,232 @@
+	else
+		movable_parent.balloon_alert_to_viewers("шляпа спадает!")
+
+	if(!istype(parent, /obj/item/clothing))
```
у нас скорее всего есть изхелпер для одежды. Используй его.

dageavtobusnick commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -7,6 +7,12 @@
+/datum/outfit/admin/post_equip(mob/living/carbon/human/H, visualsOnly)
+	. = ..()
+	if(ismodcontrol(H.back))
+		var/obj/item/mod/control/C = H.back
```
имя переменной

dageavtobusnick commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -7,6 +7,12 @@
 
+/datum/outfit/admin/post_equip(mob/living/carbon/human/H, visualsOnly)
+	. = ..()
+	if(ismodcontrol(H.back))
```
ранний ретурн

dageavtobusnick commented on code/datums/spells/summonitem.dm:
```diff
@@ -86,6 +86,10 @@
 					if(is_type_in_typecache(item_to_retrieve.loc, blacklisted_summons))
 						break
 					item_to_retrieve = item_to_retrieve.loc
+					if(istype(item_to_retrieve, /obj/item/storage/backpack/modstorage))
```
если используется не 1 раз - лучше сделать изхелпер

dageavtobusnick commented on code/datums/components/hat_stabilizer.dm:
```diff
@@ -0,0 +1,232 @@
+	apparel.detach_clothing_traits(attached_hat.clothing_traits)
+	apparel.flags_cover = former_flags
+	apparel.visor_flags_cover = former_visor_flags
+	apparel.update_appearance()
```
тут точно нужен полный апдейт всей внешки?

dageavtobusnick commented on code/datums/wires/wires.dm:
```diff
@@ -507,3 +507,15 @@
+	var/remaining_pulses = 3
+
+	for(var/wire in possible_wires)
+		if(prob(33))
```
ранний continue

dageavtobusnick commented on code/game/gamemodes/wizard/spellbook.dm:
```diff
@@ -426,13 +426,10 @@
 
 /datum/spellbook_entry/item/scryingorb/Buy(mob/living/carbon/human/user, obj/item/spellbook/book)
 	if(..())
-		if(!HAS_TRAIT_FROM(user, TRAIT_XRAY, SCRYING_ORB_TRAIT))
```
а тут и выше точно не нужны проверки, особенно учитывая, что источник стал менее частным?

NightDawnFox replied:
???

dageavtobusnick commented on code/game/jobs/job/job.dm:
```diff
@@ -290,6 +290,10 @@
 		qdel(gear_leftovers)
 
+	if(ismodcontrol(H.back))
+		var/obj/item/mod/control/C = H.back
```
имя переменной

dageavtobusnick commented on code/game/machinery/cryopod.dm:
```diff
@@ -374,6 +377,10 @@
 			QDEL_NULL(P.id)
 			qdel(P)
 			continue
+		if(istype(I, /obj/item/storage/backpack/modstorage)) //Best place for me to put it.
```
ну вот и второе использование. Значит изхелпер нужен

dageavtobusnick commented on code/game/machinery/cryopod.dm:
```diff
@@ -374,6 +377,10 @@
 			qdel(P)
 			continue
+		if(istype(I, /obj/item/storage/backpack/modstorage)) //Best place for me to put it.
+			var/obj/item/storage/backpack/modstorage/M = I
```
имя переменной

dageavtobusnick commented on code/game/machinery/deployable.dm:
```diff
@@ -189,6 +209,352 @@
+	/// This variable is used to tell the shield to ping it's owner when it is broke.
+	var/core_shield = FALSE
+	/// This variable is to tell the shield what it's source is.
+	var/obj/structure/dropwall_generator/source = null
```
переменные наследника после измененных переменных родителя

dageavtobusnick commented on code/game/machinery/deployable.dm:
```diff
@@ -189,6 +209,352 @@
+	anchored = TRUE
+	protected = TRUE
+	addtimer(CALLBACK(src, PROC_REF(power_out)), uptime)
+	timer_overlay_proc(uptime/10)
```
```suggestion
	timer_overlay_proc(uptime / 10)
```

dageavtobusnick commented on code/game/machinery/deployable.dm:
```diff
@@ -189,6 +209,352 @@
+/obj/structure/dropwall_generator/emp_act(severity)
+	..()
+	if(protected)
+		for(var/obj/structure/barricade/dropwall/O in connected_shields)
```
имя переменной

dageavtobusnick commented on code/game/machinery/deployable.dm:
```diff
@@ -189,6 +209,352 @@
+	if(protected)
+		for(var/obj/structure/barricade/dropwall/O in connected_shields)
+			O.emp_act(severity)
+	else
```
тут и выше лучше сделать через ранний ретурн, а не через else

dageavtobusnick commented on code/game/machinery/rechargestation.dm:
```diff
@@ -150,21 +150,21 @@
-				H.set_nutrition(min(H.nutrition + recharge_speed_nutrition, 450))
+	SEND_SIGNAL(occupant, COMSIG_PROCESS_BORGCHARGER_OCCUPANT, recharge_speed, repairs)
+	if(isrobot(occupant))
+		var/mob/living/silicon/robot/R = occupant
```
имя переменной

dageavtobusnick commented on code/game/machinery/rechargestation.dm:
```diff
@@ -150,21 +150,21 @@
+		if(R.cell)
+			R.cell.charge = min(R.cell.charge + recharge_speed, R.cell.maxcharge)
+	else if(ishuman(occupant))
+		var/mob/living/carbon/human/H = occupant
```
и тут

dageavtobusnick commented on code/game/objects/items/bodybag.dm:
```diff
@@ -327,3 +327,348 @@
+
+/obj/structure/closet/body_bag/environmental/update_overlays()
+	. = ..()
+	if(opened)
```
ранний ретурн

dageavtobusnick commented on code/game/objects/items/bodybag.dm:
```diff
@@ -327,3 +327,348 @@
+		if(istype(loc, /obj/machinery/disposal))
+			return ..()
+		bust_open()
+	else
```
лучше через инверт и ранний ретурн

dageavtobusnick commented on code/game/objects/items/weapons/implants/implant_mod_pathfinder.dm:
```diff
@@ -0,0 +1,182 @@
+	module.mod.cut_overlay(jet_icon)
+	module.mod.transform = matrix()
+	UnregisterSignal(module.mod, COMSIG_MOVABLE_MOVED)
+	if(!successful)
```
ранний ретурн

dageavtobusnick commented on code/game/objects/items/weapons/implants/implant_mod_pathfinder.dm:
```diff
@@ -0,0 +1,182 @@
+	if(tries >= 5)
+		set_path(null)
+	var/target = get_turf(imp_in)
+	var/mob/living/carbon/human/H = imp_in
```
имя переменной

dageavtobusnick commented on code/game/objects/items/weapons/implants/implant_mod_pathfinder.dm:
```diff
@@ -0,0 +1,182 @@
+/obj/item/implant/mod/proc/mod_step() //Step,increase tries if failed
+	if(!path || !length(path))
+		return FALSE
+	for(var/obj/machinery/door/D in range(2, module.mod))
```
имя переменной

dageavtobusnick commented on code/game/objects/items/weapons/storage/lockbox.dm:
```diff
@@ -170,6 +170,18 @@
+
+/obj/item/storage/lockbox/research/modsuit/emp_act(severity) //I want emp to get around it, it's not a gun, I just want people not to always make sec / med modsuits.
+	. = ..()
+	if(prob(50 / severity) && !broken)
```
ранний ретурн

dageavtobusnick commented on code/modules/clothing/chameleon/chameleon_actions_subtypes.dm:
```diff
@@ -154,3 +154,33 @@
+
+/datum/action/item_action/chameleon/change/modsuit/update_item(obj/item/picked_item)
+	. = ..()
+	if(ismodcontrol(target))
```
ранний ретурн

dageavtobusnick commented on code/modules/clothing/chameleon/chameleon_actions_subtypes.dm:
```diff
@@ -154,3 +154,33 @@
+/datum/action/item_action/chameleon/change/modsuit/update_item(obj/item/picked_item)
+	. = ..()
+	if(ismodcontrol(target))
+		var/obj/item/mod/control/C = target
```
имя переменной

dageavtobusnick commented on code/modules/clothing/chameleon/chameleon_actions_subtypes.dm:
```diff
@@ -154,3 +154,33 @@
+
+/datum/action/item_action/chameleon/change/modsuit/select_look(mob/user)
+	if(ismodcontrol(target))
+		var/obj/item/mod/control/C = target
```
имя переменной

dageavtobusnick commented on code/modules/clothing/chameleon/chameleon_actions_subtypes.dm:
```diff
@@ -154,3 +154,33 @@
+			tosignal.RegisterSignal(C, COMSIG_MOD_ACTIVATE, TYPE_PROC_REF(/obj/item/mod/module/chameleon, return_look))
+
+/datum/action/item_action/chameleon/change/modsuit/select_look(mob/user)
+	if(ismodcontrol(target))
```
ранний ретурн

dageavtobusnick commented on code/modules/clothing/upgrade_modules/hardsuit_taser_proof_module/hardsuit_taser_proof.dm:
```diff
@@ -29,11 +29,11 @@
 			new /obj/item/hardsuit_taser_proof/ert_locked(src)
 
 /obj/item/hardsuit_taser_proof/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = ITEM_ATTACK)
+	. = FALSE
```
не поддаваяся тяжелому тгшному заболеванию. Вариант до изменения был лучше

dageavtobusnick commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -533,6 +533,11 @@
 			if(helmet?.examine_extensions)
 				have_hud_exam |= helmet.examine_extensions
 
+		if(istype(H.head, /obj/item/clothing/head/mod))
```
аналогично. Если используется не 1 раз, лучше сделать изхелпер

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,312 @@
+		playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE, SILENCED_SOUND_EXTRARANGE)
+	if(part_datum.can_overslot)
+		var/obj/item/overslot = wearer.get_item_by_slot(part.slot_flags)
+		if(istype(overslot, /obj/item/clothing))
```
изхелпер

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,312 @@
+				continue
+			choose_deploy(user)
+			break
+	else
```
лучше ретурн сделать

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,312 @@
+		if(!force_deactivate)
+			balloon_alert(user, "уже [active ? "отключается" : "включается"]!")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE, SILENCED_SOUND_EXTRARANGE)
+		return FALSE
```
проверки бы вынести в отдельный прок

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,312 @@
+/obj/item/mod/control/proc/delayed_seal_part(obj/item/clothing/part)
+	. = FALSE
+	var/datum/mod_part/part_datum = get_part_datum(part)
+	if(do_after(wearer, activation_step_time, wearer, MOD_ACTIVATION_STEP_FLAGS, max_interact_count = 1, extra_checks = CALLBACK(src, PROC_REF(get_wearer))))
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,312 @@
+
+/obj/item/mod/control/proc/delayed_activation()
+	. = FALSE
+	if(do_after(wearer, activation_step_time, wearer, MOD_ACTIVATION_STEP_FLAGS, extra_checks = CALLBACK(src, PROC_REF(get_wearer))))
```
и тут

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,312 @@
+				continue
+			module.on_part_activation()
+			module.part_activated = TRUE
+	else
```
и тут

dageavtobusnick commented on code/modules/mod/mod_construction.dm:
```diff
@@ -0,0 +1,387 @@
+				construction_step = START_STEP
+
+		if(SCREWED_CORE_STEP)
+			if(istype(part, /obj/item/mod/construction/helmet)) //Construct
```
изхелперы в подобных местах во всем проке, если используются не единажды.

NightDawnFox replied:
этот единожды

dageavtobusnick replied:
Если большая часть с изхелперами, лучше и остатки сделать, на будущее.

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,777 @@
+	. = ..()
+	if(slot == ITEM_SLOT_BACK)
+		set_wearer(user)
+	else if(wearer)
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,777 @@
+	if(ismecha(carbon_mob.loc))
+		return
+
+	if(!HAS_TRAIT(carbon_mob, TRAIT_RESTRAINED) && !carbon_mob.stat)
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,777 @@
+		install(attacking_item, user)
+		SEND_SIGNAL(src, COMSIG_MOD_MODULE_ADDED, user)
+		return ATTACK_CHAIN_PROCEED
+	else if(istype(attacking_item, /obj/item/mod/core))
```
изхелперы. + не используй else, если ты итак возвращаешь значение

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,777 @@
+		if(bag)
+			bag.forceMove(user)
+			bag.show_to(user)
+	else
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,777 @@
+
+/obj/item/mod/control/click_alt(mob/user)
+	. = ..()
+	if(ishuman(user) && Adjacent(user) && !user.incapacitated(FALSE, TRUE) && bag)
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,777 @@
+	if(part in modules)
+		uninstall(part)
+		return
+	if(part in get_parts())
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/mod_core.dm:
```diff
@@ -0,0 +1,363 @@
+			mod.wearer.throw_alert("mod_charge", /atom/movable/screen/alert/lowcell, 1)
+		if(0.25 to 0.5)
+			mod.wearer.throw_alert("mod_charge", /atom/movable/screen/alert/lowcell, 2)
+		if(0.01 to 0.25)
```
что это за цифры? дефайнизируй

dageavtobusnick commented on code/modules/mod/mod_core.dm:
```diff
@@ -0,0 +1,363 @@
+			return "bad"
+		if(0.33 to 0.66)
+			return "average"
+		if(0.66 to INFINITY)
```
и тут

dageavtobusnick commented on code/modules/mod/mod_core.dm:
```diff
@@ -0,0 +1,363 @@
+	mod.update_charge_alert()
+
+/obj/item/mod/core/standard/on_attackby(obj/item/attacking_item, mob/user, params)
+	if(istype(attacking_item, /obj/item/stock_parts/cell))
```
изхелпер

dageavtobusnick commented on code/modules/mod/mod_core.dm:
```diff
@@ -0,0 +1,363 @@
+	switch(round(charge_amount() / max_charge_amount(), 0.01))
+		if(-INFINITY to 0.33)
+			return "bad"
+		if(0.33 to INFINITY)
```
дефайны тут и выше. А еще это копирование, как и у стандарта. Нельзя перенести в родитель или вынести в отдельные проки?

NightDawnFox replied:
Копирование это допущение, созданное для возможности более тонкой настройки при добавлении новых ядер, которых на тг минимум 4 штуки (с уникальной внешкой нехватки энергии), их я пока не портировал. Пусть будет..

dageavtobusnick commented on code/modules/mod/mod_theme.dm:
```diff
@@ -0,0 +1,2288 @@
+		/obj/item/gun,
+	)
+	variants = list(
+		"debug" = list(
```
подобные строки в ключах списков нужно дефайнизировать

dageavtobusnick commented on code/modules/wiremod/shell/module.dm:
```diff
@@ -0,0 +1,314 @@
+		module_to_select.possible_options = modules_list
+		if(length(module_to_select.possible_options))
+			module_to_select.set_value(module_to_select.possible_options[1])
+	else if(istype(old_loc, /obj/item/mod/control))
```
ранний ретурн и иххелперы

dageavtobusnick commented on code/modules/research/designs/wiremod_designs.dm:
```diff
@@ -592,14 +592,15 @@
-// 	)
+/datum/design/mod_module_shell
+	id = "module_shell"
+	req_tech = list("materials" = 2, "programming" = 2)
```
техи дефайнизированы. используй дефайны

dageavtobusnick commented on code/modules/reagents/chemistry/reagents/toxins.dm:
```diff
@@ -454,8 +454,15 @@
-				to_chat(H, span_danger("Ваш[GEND_A_E_I(H.head)] [H.head.declent_ru(NOMINATIVE)] плавится!"))
-				qdel(H.head)
+				if(istype(H.head, /obj/item/clothing/head/mod) && ismodcontrol(H.back))
+					var/obj/item/mod/control/C = H.back
```
имя переменной

dageavtobusnick commented on code/modules/reagents/chemistry/reagents/toxins.dm:
```diff
@@ -454,8 +454,15 @@
 			if(H.head && !(H.head.resistance_flags & ACID_PROOF))
-				to_chat(H, span_danger("Ваш[GEND_A_E_I(H.head)] [H.head.declent_ru(NOMINATIVE)] плавится!"))
-				qdel(H.head)
+				if(istype(H.head, /obj/item/clothing/head/mod) && ismodcontrol(H.back))
```
Про исхелперы я устал писать. Очевидно что они неоднократно используются в коде. На каждый компонент мода скорее всего нужен будет изхелпер и земенить изтайпы на изхелперы там, где они используются.

dageavtobusnick commented on code/modules/projectiles/projectile/bullets.dm:
```diff
@@ -192,13 +192,21 @@
+	if(!leaves_fire_trail)
+		return
+	var/turf/location = get_turf(src)
+	if(location)
```
ранний ретурн

dageavtobusnick commented on code/modules/projectiles/projectile/bullets.dm:
```diff
@@ -207,6 +215,29 @@
+/obj/projectile/bullet/incendiary/fire/on_hit(atom/target, blocked = 0, pierce_hit)
+	. = ..()
+	var/turf/location = get_turf(target)
+	if(location && !location.density)
```
и тут

dageavtobusnick commented on code/modules/projectiles/guns/energy/special.dm:
```diff
@@ -869,6 +869,94 @@
+	if(blocked >= 100)
+		return ..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/L = target
```
имя переменной

dageavtobusnick commented on code/modules/projectiles/guns/energy/special.dm:
```diff
@@ -869,6 +869,94 @@
+
+/obj/effect/warp_effect/vortex_blast/Initialize(mapload, target)
+	. = ..()
+	var/matrix/M = matrix() * 0.5
```
и тут

dageavtobusnick commented on code/modules/mod/modules/_modules.dm:
```diff
@@ -0,0 +1,600 @@
+/obj/item/mod/module/proc/on_device_deletion(datum/source)
+	SIGNAL_HANDLER
+
+	if(source == device)
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/modules/_modules.dm:
```diff
@@ -0,0 +1,600 @@
+	if(core)
+		. += span_notice("К модулю прикреплен[GEND_A_O_Y(core)] [core.declent_ru(NOMINATIVE)]. \
+						[core_removable ? "Мо[PLUR_JET_GUT(core)] быть откручен[GEND_A_O_Y(core)]." : "Конструкция модуля не позволяет извлечь ядро."]")
+	else
```
тут тоже

dageavtobusnick commented on code/modules/mod/modules/module_kinesis.dm:
```diff
@@ -0,0 +1,340 @@
+	UnregisterSignal(source, COMSIG_MOVABLE_IMPACT)
+	if(!isobj(source))
+		return
+	var/obj/S = source
```
имя переменной

dageavtobusnick commented on code/modules/mod/modules/modules_antag.dm:
```diff
@@ -0,0 +1,1038 @@
+	var/calculated_protection = round((core.get_strength() / 3))
+	var/calculated_charge = round((core.get_strength() / 9))
+	var/calculated_range =  min(round((core.get_strength() / 20)), maximum_zap_range) //limit of 7
+	var/calculated_damage = round((core.get_strength() / 7))
```
магические числа надо дефайнизировать

dageavtobusnick commented on code/modules/mod/modules/modules_antag.dm:
```diff
@@ -0,0 +1,1038 @@
+	charge_recovery = calculated_charge
+	zap_range = calculated_range
+	shock_damage = calculated_damage
+	if(core.get_strength() > 200)
```
и тут

dageavtobusnick commented on code/modules/mod/modules/modules_antag.dm:
```diff
@@ -0,0 +1,1038 @@
+			M.electrocute_act(shock_damage, owner)
+			M.Knockdown(3 SECONDS)
+	else if(isliving(hitby))
+		var/mob/living/M = hitby
```
имена переменных и ранние ретурны

dageavtobusnick commented on code/modules/mod/modules/modules_antag.dm:
```diff
@@ -0,0 +1,1038 @@
+	if(blocked >= 100)
+		return 0
+	var/turf/firer_turf = get_turf(firer)
+	if(isliving(target))
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/modules/modules_antag.dm:
```diff
@@ -0,0 +1,1038 @@
+		if("security")
+			choosed_name = "модели \"Страж\""
+			choosed_skin = "security"
+		if("engineering")
```
выборы дефайнизировать. Мне кажется названия те же, что и в темах, поэтому дефайны можно использовать те же

dageavtobusnick commented on code/modules/mod/modules/modules_antag.dm:
```diff
@@ -0,0 +1,1038 @@
+			choosed_name = "модели \"Искра\""
+			choosed_skin = "engineering"
+
+	sleep(25)
```
addTimer и 2.5 SECONDS

dageavtobusnick commented on code/modules/mod/modules/modules_engineering.dm:
```diff
@@ -0,0 +1,480 @@
+
+/obj/projectile/tether/on_hit(atom/target)
+	. = ..()
+	if(firer && isliving(firer))
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/modules/modules_engineering.dm:
```diff
@@ -0,0 +1,480 @@
+	ricochet_chance = 0
+
+/obj/projectile/tether/proc/make_chain()
+	if(firer)
```
и тут

dageavtobusnick commented on code/modules/mod/modules/modules_engineering.dm:
```diff
@@ -0,0 +1,480 @@
+/obj/projectile/tether/on_hit(atom/target)
+	. = ..()
+	if(firer && isliving(firer))
+		var/mob/living/L = firer
```
имя переменной

dageavtobusnick commented on code/modules/mod/modules/modules_general.dm:
```diff
@@ -0,0 +1,1105 @@
+/obj/item/mod/module/storage/on_uninstall(deleting = FALSE)
+	. = ..()
+	if(!deleting)
+		for(var/obj/I in bag.contents)
```
имя переменной. А еще. Почему не atom и не as anything? мы же по идее должны все вытрехнуть.

NightDawnFox replied:
undefined proc: "forceMove" on /atom

dageavtobusnick replied:
Тогда не нужно

dageavtobusnick commented on code/modules/mod/modules/modules_general.dm:
```diff
@@ -0,0 +1,1105 @@
+
+/obj/item/storage/backpack/modstorage/update_viewers()
+	for(var/_M in mobs_viewing)
+		var/mob/M = _M
```
as anything и нормальное имя

NightDawnFox replied:
code/modules/mod/modules/modules_general.dm:183: 'as anything' used in a typeless for loop. This doesn't do anything and should be removed.

dageavtobusnick replied:
for(var/mob/имя_переменной as anything in mobs_viewing)

dageavtobusnick replied:
Если нужна более гибкая настройка, то запихивай в переменные

dageavtobusnick commented on code/modules/mod/modules/modules_general.dm:
```diff
@@ -0,0 +1,1105 @@
+/obj/item/mod/module/flashlight/configure_edit(key, value)
+	switch(key)
+		if("light_color")
+			value = input(usr, "Выберите цвет фонаря", "Цвет фонаря") as color|null // на тгуи ввод бы перевести
```
у нас есть tgui инпут для цвета. Используй его

dageavtobusnick commented on code/modules/mod/modules/modules_general.dm:
```diff
@@ -0,0 +1,1105 @@
+		helmet.flags_cover &= ~(HEADCOVERSMOUTH)
+		helmet.visor_flags_cover &= ~(HEADCOVERSMOUTH)
+	var/obj/item/clothing/mask = mod.get_part_from_slot(ITEM_SLOT_MASK)
+	if(istype(mask))
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/modules/modules_general.dm:
```diff
@@ -0,0 +1,1105 @@
+		helmet.flags_cover |= former_helmet_flags
+		helmet.visor_flags_cover |= former_visor_helmet_flags
+	var/obj/item/clothing/mask = mod.get_part_from_slot(ITEM_SLOT_MASK)
+	if(istype(mask))
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/modules/modules_maint.dm:
```diff
@@ -0,0 +1,242 @@
+	update_icon(UPDATE_ICON_STATE)
+
+/obj/item/stamp/mod/update_icon_state()
+	if(icon_state == "stamp-ok")
```
лучше завести у объекты переменную и изменять ее, и на основе ее менять икон стейт, а не проверять на икон стейт.

dageavtobusnick commented on code/modules/mod/modules/modules_security.dm:
```diff
@@ -0,0 +1,651 @@
+	. = ..()
+	if(!looker || !creature)
+		return INITIALIZE_HINT_QDEL
+	modsuit_image = image(icon = icon, loc = src, icon_state = real_icon_state, layer = ABOVE_ALL_MOB_LAYER, pixel_x = ((creature.x - looker.x) * 32), pixel_y = ((creature.y - looker.y) * 32))
```
у нас есть дефайн для стандартного размера иконки

dageavtobusnick commented on code/modules/mod/modules/modules_security.dm:
```diff
@@ -0,0 +1,651 @@
+/obj/item/mod/module/anomaly_locked/vortex_shotgun/update_core_powers()
+	if(!core)
+		return
+	use_energy_cost = DEFAULT_CHARGE_DRAIN * 1000
```
магические цифры лучше дефайнизировать

dageavtobusnick commented on code/modules/mod/modules/modules_security.dm:
```diff
@@ -0,0 +1,651 @@
+	if(!gun.chambered.BB)
+		gun.chambered.newshot()
+
+	if(!drain_power(use_energy_cost)) //no shoot when no energy
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+			return
+		dropped_crate.forceMove(target_turf)
+		drain_power(use_energy_cost)
+	else
```
ранние ретурны

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+		return
+	if(!mod.wearer.Adjacent(target))
+		return
+	if(ismineralturf(target))
```
ранний ретурн

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+				part.slowdown -= speed_added / 5
+	else if(is_type_in_typecache(mod.wearer.loc, keep_turfs))
+		return
+	else
```
ломаем лесенки, используем ранние ретурны

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+	if(locker.anchored || locker.move_resist >= MOVE_FORCE_OVERPOWERING)
+		return
+	playsound(locker, 'sound/effects/gravhit.ogg', 75, TRUE)
+	locker.throw_at(get_step_towards(mod.wearer, target), range = 7, speed = 3, force = MOVE_FORCE_WEAK, \
```
лучше сделать каждую переменную на своей строке в подобных моментах. Такое есть при добавлении компонентов. проще чистается.

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+	playsound(src, 'sound/items/modsuit/loader_launch.ogg', 75, TRUE)
+	var/angle = get_angle(mod.wearer, target)
+	mod.wearer.transform = mod.wearer.transform.Turn(mod.wearer.transform, angle)
+	mod.wearer.throw_at(get_ranged_target_turf_direct(mod.wearer, target, power), \
```
тут тоже

NightDawnFox replied:
тут буквально некуда

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+/obj/structure/mining_bomb/proc/boom(atom/movable/firer)
+	visible_message(span_danger("[capitalize(declent_ru(NOMINATIVE))] взрывается!"))
+	playsound(src, 'sound/magic/magic_missile.ogg', 200, vary = TRUE)
+	for(var/turf/T in circle_view_turfs(src, 2))
```
а могут быть не турфы? если нет, то as anything. + нормальное имя для переменной.

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+	visible_message(span_danger("[capitalize(declent_ru(NOMINATIVE))] взрывается!"))
+	playsound(src, 'sound/magic/magic_missile.ogg', 200, vary = TRUE)
+	for(var/turf/T in circle_view_turfs(src, 2))
+		if(ismineralturf(T))
```
ранний continue

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,672 @@
+	. = ..()
+	if(!.)
+		return
+	if(istype(mod.wearer.pulling, /obj/structure/closet))
```
изхелпер

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into eds2

NightDawnFox committed:
пиздец блядь

NightDawnFox committed:
фиксы линтера

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -353,21 +358,22 @@
+		return
+	bag.forceMove(user)
+	bag.show_to(user)
+	playsound(loc, "rustle", 50, TRUE, -5)
```
Тут скорее всего должен быть какой то SFX

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -199,23 +202,25 @@
-					playsound(src, 'sound/machines/scanbuzz.ogg', 25, FALSE, SILENCED_SOUND_EXTRARANGE)
-					return
-			if(!carbon_mob.temporarily_remove_item_from_inventory(src, silent = TRUE))
+	playsound(loc, "rustle", 50, TRUE, -5)
```
И тут

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -270,15 +278,16 @@
-				continue
-			module.deactivate(display_message = FALSE)
+		return
+	//else
```
К чему этот комментарий

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -34,15 +34,17 @@
-			break
+		return
+
+	// else
```
К чему комментарий

dageavtobusnick commented on code/modules/mod/modules/_modules.dm:
```diff
@@ -529,15 +530,16 @@
-		if(!core_removable)
-			. += span_notice("Конструкция модуля не позволяет извлечь ядро.")
+		return
+	//else
```
Комментарий

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -299,9 +301,13 @@
+	mod.wearer.throw_at(
+		get_ranged_target_turf_direct(mod.wearer, target, power),
+		range = power,
+		speed = max(round(0.2*power), 1),
```
Пробелы до и после *

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,784 @@
+	if(HAS_TRAIT(carbon_mob, TRAIT_RESTRAINED) || carbon_mob.stat) //restrained or unconsious
+		return
+
+	playsound(loc, "rustle", 50, TRUE, -5)
```
```suggestion
	playsound(loc, SFX_RUSTLE, 50, TRUE, -5)
```

dageavtobusnick commented on code/modules/mod/mod_control.dm:
```diff
@@ -0,0 +1,784 @@
+		return
+	bag.forceMove(user)
+	bag.show_to(user)
+	playsound(loc, "rustle", 50, TRUE, -5)
```
```suggestion
	playsound(loc, SFX_RUSTLE, 50, TRUE, -5)
```

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,321 @@
+			module.on_part_activation()
+			module.part_activated = TRUE
+		return
+	//else
```
```suggestion
```

dageavtobusnick commented on code/modules/mod/mod_activation.dm:
```diff
@@ -0,0 +1,321 @@
+			break
+		return
+
+	// else
```
```suggestion
```

dageavtobusnick commented on code/modules/mod/modules/_modules.dm:
```diff
@@ -0,0 +1,602 @@
+		. += span_notice("К модулю прикреплен[GEND_A_O_Y(core)] [core.declent_ru(NOMINATIVE)]. \
+						[core_removable ? "Мо[PLUR_JET_GUT(core)] быть откручен[GEND_A_O_Y(core)]." : "Конструкция модуля не позволяет извлечь ядро."]")
+		return
+	//else
```
```suggestion
```

dageavtobusnick commented on code/modules/mod/modules/modules_supply.dm:
```diff
@@ -0,0 +1,687 @@
+	mod.wearer.throw_at(
+		get_ranged_target_turf_direct(mod.wearer, target, power),
+		range = power,
+		speed = max(round(0.2*power), 1),
```
```suggestion
		speed = max(round(0.2 * power), 1),
```

dageavtobusnick committed:
Apply suggestions from code review

LussatTheVulpa commented:
Я являюсь автором спрайтов модсьюта пилота "brigpilot" в следующих файлах:

icons/obj/clothing/modsuit/mod_clothing.dmi

icons/mob/clothing/modsuit/mod_clothing.dmi

icons/mob/clothing/modsuit/species/vulpkanin/mod_clothing.dmi

icons/mob/clothing/modsuit/species/vox/mod_clothing.dmi

icons/mob/clothing/modsuit/species/unathi/mod_clothing.dmi

icons/mob/clothing/modsuit/species/tajaran/mod_clothing.dmi

icons/mob/clothing/modsuit/species/grey/mod_clothing.dmi

icons/mob/clothing/modsuit/species/drask/mod_clothing.dmi

Данные работы были добавлены в этот репозиторий в коммите 7f4f6b252442422370f8e214b51a6a3025537a71 https://github.com/ss220-space/Paradise/pull/7500/changes/7f4f6b252442422370f8e214b51a6a3025537a71 пользователем NightDawnFox.

Я не давал письменного или явного согласия на лицензирование моих работ под лицензией CC BY-SA 3.0 данного репозитория. Моё неформальное разрешение на использование не распространялось на их публичное распространение на этих условиях.

Я официально требую удаления моих работ в течение 7 дней.

В случае если материалы не будут удалены, я буду вынужден подать уведомление о нарушении авторских прав по закону DMCA в GitHub.

LussatTheVulpa.
